### PR TITLE
Let's shake and bake, baby [DRUG OVERHAUL PART 1]

### PR DESCRIPTION
**Pull request by kilkun**
:cl:
add: Added a new drug, Amphetamine.
code: Methamphetamine and Crank are now children of amphetamine.
tweak: Meth is a lot deadlier and less effective
tweak: Crank is a LOT deadlier
balance: Crafty assistants are now capable of making crank, but be warned my ghetto chemists: it comes at a hefty cost to your health as opposed to pure pharmaceutical meth.
/:cl:

This is the first part of an overarching series of PRs that I'm going to be making regarding drugs, a sorely neglected and horribly balanced part of the codebase. This one mainly focuses on amphetamines and substituted amphetamines, specifically shifting them towards a more realistic implementation.
The particulars:
-Crank can be made with supplies found around the station and with the help of botany.
-Meth and crank carry a risk of cardiac arrest, crank is 10x more likely to result in a cardiac event. Overdose increases the likelihood of a heart attack x100
-Meth and crank now cause psychosis.
-Although it's currently not capable of being synthesized, severe amphetamine withdrawals will also result in psychosis.

I'm open to specifically modifying the numbers, but the mechanics are important.

It also rebalances these chemicals. In particular, meth is not a super anti-stun, but it retains effectiveness against retaining consciousness. I realize taking away the super anti-stuns of these chemicals and making them more unpredictably lethal seems like a sour move, but the goal of this drug overhaul is to reflect the actual nature of the pharmacology of these substances. You can no longer chemistry your way out of a taser fight, and you have a small chance to keel over if you try. Don't worry, though, the replacement anti-stuns are fully planned. You'll have to deal with some drawbacks, however. No more dosing 18u meth/crank freebase pills every 17 tics for essentially NOSTUN

I have big plans for this. Consider these initial, small changes dipping my toe in the water. If the player base responds positively to the changes, I will resume this overhaul and that's when we get to the fun stuff like research chemicals, seizures, dissociatives, etc.

### CODE REVIEW

**Comment by Mark9013100** [comment]
ADD HEROIN

File: code/modules/reagents/chemistry/recipes/drugs.dm [hunk 1]
```diff
@@ -4,13 +4,7 @@
...
 	required_reagents = list("mercury" = 1, "sugar" = 1, "lithium" = 1)
 
-/datum/chemical_reaction/crank
-	name = "Crank"
-	id = "crank"
-	results = list("crank" = 5)
-	required_reagents = list("diphenhydramine" = 1, "ammonia" = 1, "lithium" = 1, "sacid" = 1, "welding_fuel" = 1)
-	mix_message = "The mixture violently reacts, leaving behind a few crystalline shards."
-	required_temp = 390
+
```
Comment (praisenarsie) [inline code/modules/reagents/chemistry/recipes/drugs.dm]: so much whitespace

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -337,3 +215,172 @@
...
+	. = 1
+
+
+/datum/reagent/drug/amphetamine/addiction_act_stage4(mob/living/M)
+	if(prob(30))
+		M.adjustStaminaLoss(2, 0)
+		to_chat(M, "<span class='boldannounce'>You're not feeling good at all! You really need some [name].</span>")
+	return
+
+
```
Comment (praisenarsie) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: there's lots of whitespace in this file

**Comment by Supermichael777** [comment]
Making drugs children of other drugs is a bad idea

most drugs have a custom onmoblife an are dependent on an unbroken onmoblife to function corectly

Having an unnecessary parent is just as bad as missing a parent type you need. its what broke  sleepypens for a year

**Comment by kilkun** [comment]
@Mark9013100 
Opiod receptors really have no good analogue in the game.

That being said, I was thinking about maybe overhauling how metabolization happens (shooting isn't the same as swallowing a pill isn't the same as snorting isn't the same as vaporizing) and I might add in something along the lines to make euphoria/pain killing relevant in game mechanics. I'm pretty dissatisfied with krokodil and morphine.

Also U47700 is totally on the list of drugs to add in because holy shit that caustic behavior is so SS13 it'd be a crime to exclude it.

@Supermichael777 
I mean the chemistry system is anything but ideal, if this breaks I'll change it but right now it was an easy way to implement "dirty" meth as crank without copy pasting.

**Comment by Supermichael777** [comment]
in addition its typically bad practice to have functional objects as children of other functional objects if their effects are different.  Make a super type and have all three be independent children that can be balanced separately

**Comment by kilkun** [comment]
Amphetamine is the super type.

I mean ideally, phenethylamine would be the super type, with amphetamine being a subtype and crank and meth being a subtype of amphetamine, but that would just create height in the inheritance tree for no good reason because I have no plans to add methylphenidate or any other non-amphetamine substituted phenethylamine.

**Comment by DoctoreMad** [comment]
>because I have no plans to add methylphenidate or any other non-amphetamine substituted phenethylamine.

And if someone does?

**Comment by kilkun** [comment]
we can burn that bridge when we cross it, but for now it's adding definitions and increasing hierarchy complexity for very very very very little payoff so i've occluded it.

**Comment by kilkun** [comment]
yes at some point racemic amphetamine will be able to be synthesized but I need to find a way to make it balanced and allow for maintenance rat assistants to synthesize it without stealing chemistry gear. the goal being, amphetamine as a marginal improvement over nicotine with slightly more dangers associated with it.

as it currently stands, I haven't yet thought of how I'm going to do that so it's going to be sitting pretty until the next step in the project.

**Comment by Supermichael777** [comment]
your injecting real world logic into code parent logic what I'm saying is that you should have a super-type like /stimulant that takes its stun and sleep reduction from vars and checks for heart attacks and then have meth and such be 3 independently balanceable chems based of that parent.  

Say for example I think amphetamines need more stamina regen but don't want to adjust meth and crank. your setup makes that hard and hacky (i would need to give meth a stamina malice). With a proper super-type setup I only need to adjust a var in amphetamine. 

additionally this would cut out one of the massive onmoblifes.

so instead of being `/amphetamine/methamphetamine/crank`
it ends up as 
```
/stimulant/crank
/stimulant/amphetamine
/stimulant/methamphetamine
```

**Comment by Anonmare** [comment]
I've been wanting to add reagents to a variety of commonly available station items and objects. Like phosphorus in matchsticks, iodine in camera photos/rolls and ink cartridges, copper in cables etc.

**Comment by kilkun** [comment]
@Supermichael777 I see what you're saying, but I have crank as a subtype of meth because I want them to behave the exact same way, but with different overdose/addiction thresholds and a different heart attack chance.

My approach to this might make more sense in the third part of this project when more amphetamine types are added in, like 3-FMA and 4-FMA. I'm not too attached to this method, I just wanted to cut down on copy-paste code. If a maintainer wants it changed, I'll change it, but for the sake of sanity I want to try to keep drug inheritance where it makes sense.

**Comment by DoctoreMad** [comment]
>WHERE IS ALL THE CABLE

Oh a guy was doing ghetto chemistry, he had to get a handlful.

>OH GREAT

I don't actually oppose it, for reference.

**Comment by kilkun** [comment]
gives sec a reason to crack down on assistants making the stuff :thinking:

**Comment by DoctoreMad** [comment]
Implying security needs a reason to crack down on assistants

**Comment by optimumtact** [comment]
>If the player base responds positively to the changes

lol

**Comment by ExcessiveUseOfCobblestone** [comment]
Why are they a child of X when they share nothing of X?

**Comment by kilkun** [comment]
crank shares almost everything with meth besides a few modified variables. amphetamine isn't finished yet, but by the time this project is done, the uniqueness meth has will be cut down greatly. that's why.

**Comment by ExcessiveUseOfCobblestone** [comment]
oh wow you responded right after I corrected myself fug

**Comment by Bawhoppen** [comment]
To say this is an overhaul is kind of an exaggeration.
(Only 175 lines changed)

**Comment by davethwave** [comment]
Well I guess I'm happy to except the new changes in this, the only problem is are those who now try meth/crank after this and suddenly die for no reason will probably complain or those who use it for nostuns and the like, though... they will likely be happy to use crank with a syringe gun with its new lethality.

**Comment by kilkun** [comment]
i'd rather be ODing on crank and being able to get a few shots in than just being blasted with neurotoxin and having to sit there watching my spaceman get toolboxed to death tbh

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -337,3 +215,172 @@
...
+	M.AdjustStun(-20, 0)
+	M.AdjustKnockdown(-20, 0)
+	M.AdjustUnconscious(-40, 0)
+	M.adjustStaminaLoss(-1, 0)//meth isn't actually that great of a stamina booster.
+	M.status_flags |= GOTTAGOREALLYFAST
+	M.Jitter(4)
+	M.adjustBrainLoss(0.25)
+	if(prob(5))
+		M.emote(pick("twitch", "shiver"))
+	if(prob(heart_attack_chance))
```
Comment (Supermichael777) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: use the heart attack disease we already have. i think you can start a disease at a higher stage than one if you want it to be faster.

**Comment by pantherpusher** [comment]
So are you going to add cocain because I wanna see botany turn into a columbian drug cartel

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: i feel like that'd be worse tbf, this is fairly clear and can be modified easier

**Comment by Qustinnus** [comment]
this is not really an overhaul and I'm not a huge fan of changes that rely on your promises of future PRs. what if you decide "haha fuck this" at like part 3 and we're left with a shitty half finished system?

**Comment by kilkun** [comment]
@Qustinnus 
there is no half finishing a system. at the very worst, i've simply diversified drugs and introduced more balance to what used to be incredibly overpowered. the only real goal of this project is to increase drug diversification (including drawbacks and health risks associated with these drugs) route of administration changing bioavailability is only a theoretical feature that I'm not terribly sure I want to undertake. this isn't like the forever-in-limbo syndicate cyborg race that I have yet to actually code into an antag role, if I don't feel up to the task of changing metabolization code, I'm not gonna.

@pantherpusher 
cocaine is indeed on the list of drugs to add.
a few others:
-PCP
-Bath salts -> a-PVP, Hexen, MDPV
-Maybe 2'-Hydroxycocaine for some kind of /vg/esque creatine-style "this'd be game breaking if it wasn't coded to kill you very shortly after you take it" gimmick antag-only chem

**Comment by Jalleo** [comment]
@kilkun Ask @Cyberboss if its possible to use the component system to reduce copypasta while removing the whole inheritance issues people are talling about here. It maybe too high of a overhead to do though.

**Comment by Jalleo** [comment]
Also please learn how to use branches. It will help you longterm.

**Comment by davethwave** [comment]
Grinding electronics or boards can give silicon I guess

**Comment by kilkun** [comment]
I know how to use branches, I have two OC repositories on github that have branches for staging new features. But its hardly ever warranted to use them here for me considering I rarely make PRs so I just use master.

**Comment by Jalleo** [comment]
Understood

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 4]
```diff
@@ -337,3 +215,168 @@
...
+	if(prob(30))
+		M.adjustStaminaLoss(2, 0)
+		to_chat(M, "<span class='boldannounce'>You're not feeling good at all! You really need some [name].</span>")
+	return
+
+
+
+//Methamphetamine
+/datum/reagent/drug/amphetamine/methamphetamine
+	name = "Methmphetamine"
```
Comment (swindly) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Methamphetamine

**Comment by kilkun** [comment]
can we merge this already its been 11 days

**Comment by optimumtact** [comment]
if nobody has merged it yet it means none of us are really interested in your change

**Comment by Qustinnus** [comment]
and it doesn't seem to really improve anything

**Comment by kilkun** [comment]
it nerfs chemistry's ability to make traitor-esque nostuns which while not abused to an extreme extent, that's more or less pre-emptive as someone who has mastered it. where's the lack of improvement there?

**Comment by optimumtact** [comment]
@tgstation/commit-access closing in 24 hours unless a maintainer shows interest in getting this in

**Comment by kilkun** [comment]
outstanding

**Pull request closed without merging**